### ISSUE TITLE
[2.7.2] MongoServerSelectionError in dev mode

### ISSUE THREAD

**Issue by hexsprite** [meteor/meteor#12037]
On Platform: MacOS 12.3.1 (arm64)

```
Got exception while polling query MongoServerSelectionError: Server selection timed out after 30000 ms
```

When trying to upgrade from 2.5.6 to 2.7.2 I started running into the following exception which happens every time I start Meteor in development mode. It never actually starts the application but fails in MongoDB server selection.

Also, I had a similar problem (see #10925) before that was solved by reverting to the old way of server selection but that's no longer possible in the new Mongo driver. Hoping for some suggestions how I might troubleshoot this.

I haven't been able to create a simple reproduction yet. I have tried to copy the `.meteor/packages` into another project but that did not reproduce the error.

Any suggestions to further troubleshoot this issue?

<details>
<summary>Full exception</summary>

```
=> Started MongoDB.                           
Got exception while polling query MongoServerSelectionError: Server selection timed out after 30000 ms
    at Timeout._onTimeout (/Users/jbb/.meteor/packages/npm-mongo/.4.3.1.1f6pha6.vrrf++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb/lib/sdam/topology.js:312:38)
    at listOnTimeout (internal/timers.js:557:17)
    at processTimers (internal/timers.js:500:7)
 => awaited here:
    at Function.Promise.await (/Users/jbb/.meteor/packages/promise/.0.12.0.z6le8i.tmj7s++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/meteor-promise/promise_server.js:56:12)
    at packages/mongo/mongo_driver.js:1092:14
    at /Users/jbb/.meteor/packages/promise/.0.12.0.z6le8i.tmj7s++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/meteor-promise/fiber_pool.js:43:40
 => awaited here:
    at Promise.await (/Users/jbb/.meteor/packages/promise/.0.12.0.z6le8i.tmj7s++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/meteor-promise/promise_server.js:60:12)
    at SynchronousCursor._nextObject (packages/mongo/mongo_driver.js:1141:38)
    at SynchronousCursor.forEach (packages/mongo/mongo_driver.js:1155:22)
    at Cursor.<computed> (packages/mongo/mongo_driver.js:940:44)
    at Cursor.MeteorCursor.<computed> [as forEach] (packages/peerlibrary_reactive-mongo/server.coffee:66:22)
    at OplogObserveDriver._runQuery (packages/mongo/oplog_observe_driver.js:738:16)
    at OplogObserveDriver._runInitialQuery (packages/mongo/oplog_observe_driver.js:667:10)
    at packages/mongo/oplog_observe_driver.js:195:10
    at packages/mongo/oplog_observe_driver.js:17:9
    at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1257:12)
    at packages/meteor.js:555:25
    at runWithEnvironment (packages/meteor.js:1320:24) {
  reason: TopologyDescription {
    type: 'ReplicaSetWithPrimary',
    servers: Map(1) { '127.0.0.1:3001' => [ServerDescription] },
    stale: false,
    compatible: true,
    heartbeatFrequencyMS: 10000,
    localThresholdMS: 15,
    setName: 'meteor',
    maxSetVersion: 2568050,
    maxElectionId: new ObjectId("7fffffff0000000000000032"),
    commonWireVersion: 13,
    logicalSessionTimeoutMinutes: 30
  }
}
```
</details>

**Comment by hexsprite** [comment]
I actually added some debugging into the mongodb driver itself and found that this is erroring out with the 2 oplog connections but not the primary db connection…. not sure why that is but the list of valid servers is never updated with the live mongo server… I can manually connect to it fine and rs.status() shows it is working correctly.

**Comment by hexsprite** [comment]
Lacking any other methodology I commented out the entire `server/` directory and to my surprise all of a sudden the app at least booted (though was missing some critical features). So then I went through and commented out each file one by one. Finally I figured out when a certain module is being imported that the problems begin. It seems to be related to related to https://github.com/montagejs/collections - which overrides certain ES6 collection methods. So it doesn't seem to be a Meteor bug so I'm closing this for now.
